tests/test_utils.py

```diff
@@ -162,65 +162,85 @@ def check_if_job_finished(job: JobDataclass):
 
 def check_if_job_runner_result_correct(job: JobDataclass):
     """Iterate over every result and check if the distribution of the measurement is correct"""
-    for i in range(len(job.results)):
-        result: ResultDataclass = job.results[i]
-        check_standard_result_data(i, job, result)
-        assert result.meta_data is not None
+
+    assert len(job.results) == 2 * len(job.deployment.programs)
```

Comment:
```suggestion
    assert len({r.program.id for r in job.results}) == 2 * len(job.deployment.programs)
```
This assertion will break easily even with expected changes/differences in some pilots.